src/frontend/src/optimizer/plan_node/stream_table_scan.rs

```diff
@@ -170,6 +162,12 @@ impl StreamTableScan {
             catalog_builder.add_column(&Field::from(col));
         }
 
+        // `yielded` column
+        // Only present for arrangement backfill.
+        if is_arrangement_backfill {
+            catalog_builder.add_column(&Field::with_name(DataType::Boolean, "yielded"));
```

Comment:
Do we need to ensure backward compatibility after adding this new column? In other words, is it possible that an arrangement backfill is triggered in old version and resume in new version?

Reply:
True, I will handle it later.

Reply:
Handled in https://github.com/risingwavelabs/risingwave/pull/17266/commits/66f44cb150cfd37f9a2e53fb40227eefd0025867.

We only need to handle the deserialization path. Because backfill jobs should fall into one of 3 categories:
1. Finished -> They won't write again to state table, no need to handle encoding path. Need to still read from state table, so must handle decoding path.
2. Not finished -> They will be cleaned after upgrade, no need to handle backwards compat at all.
3. Background ddl -> They won't be cleaned upgrade, and can potentially panic. We can let users recreate these jobs. I think it's unlikely to encounter this since stream job creation should not take a long time.